pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -123,6 +121,9 @@ func (a *AgentHosts) validateAgentHosts() field.ErrorList {
 		allErrs = append(allErrs, err...)
 	}
 
+	// Warn about BMC field misconfiguration
+	installConfig.ValidateBMCConfiguration()
```

Comment:
The main issue here it's that the validation is not performed agains the `AgentHosts.Hosts`, which is currently the authoritative list of hosts, but on the `installConfig.Config.Platform.BareMetal.Hosts` one.
I think that for clarity / maintenance the asset validation step (that usually is in the `finish()` method) should focus on its objects only

Reply:
OK, called provisioning check only in installconfig asset.